sdk/src/Core/Sampling/SamplingRule.cs

```diff
@@ -173,7 +173,6 @@ internal void Merge(SamplingRule oldRule)
         {
             Statistics.CopyFrom(oldRule.Statistics);
             Reservior.CopyFrom(oldRule.Reservior);
-            oldRule = null;
```

Comment:
I'm not following how setting oldRule = null here would assist with garbage collecting.  I may be missing something, but here are my observations:

1. SamplingRule is passed as an object ref without the `ref` keyword.  Assigning `null` here will only update the reference to this object within this function to null.  The calling code `if (oldRules.TryGetValue(key, out SamplingRule oldRule))
                {
                    newRule.Merge(oldRule);
                }` would still have a reference in the `oldRules` dictionary.
2. All of the old rules, would be cleaned up in `RuleCache.LoadRules` when `_cache = tempCache;`  replaces the existing cache
3. Sampling rule doesn't implement `IDisposable`, there are no managed resources to clean up

Feel free to point out if I'm missing something, as I could always be wrong in my assessment, but I think having it set oldRule = null is actually more misleading as it isn't cleaning up the reference as you might expect.  If this were C and I was setting a pointer, I could see how this would work.  

If you would still prefer that we leave it here, I can put it back in.

Reply:
Ah I see. Sorry I didn't follow the `Merge` usage all the way through.
Thanks for clarifying.